src/main/scala/vectorpipe/util/Graph.scala

```diff
@@ -1,158 +0,0 @@
-package vectorpipe.util
-
-import cats.data.State
-import cats.implicits._
-import spire.algebra._
-import spire.std.any._
-import spire.syntax.order._
-
-// --- //
-
-/** Adapted from the Haskell implementation, which was based on the paper
-  * ''Structuring Depth-First Search Algorithms in Haskell'' by David King
-  * and John Launchbury.
-  *
-  * `Graph` has a private constructor. To create an instance of a `Graph`, use
-  * [[Graph.fromEdges]].
-  *
-  * ===Usage===
-  * The two main methods of interest are `get`:
-  * {{{
-  * // Try to find a node value that corresponds to a given key.
-  * Graph.get: K => Option[V]
-  * }}}
-  * and `topologicalForest`. Given the disconnected Graph:
-  * {{{
-  *    5      1
-  *   / \    / \
-  *  6   7  2   3
-  *   \ /    \ /
-  *    8      4
-  * }}}
-  * `topologicalForest` converts this into a [[Forest]] of topologically
-  * sorted spanning [[Tree]]s:
-  * {{{
-  * val g: Graph = ...  // From above.
-  *
-  * g.topologicalForest.foreach(t => println(t.pretty))
-  *
-  * // The above prints:
-  * └── 5
-  *     ├── 6
-  *     │   └── 8
-  *     │       └── ∅
-  *     └── 7
-  *         └── ∅
-  * └── 1
-  *     ├── 2
-  *     │   └── 4
-  *     │       └── ∅
-  *     └── 3
-  *         └── ∅
-  * }}}
-  *
-  * @constructor Use `Graph.fromEdges` instead of this.
-  */
-class Graph[K: Order, V] private (
```

Comment:
Seems like a lot of work went into this and some of the other data structures/code. Should this go somewhere for preservation, in case we need it later?

Reply:
ScalaZ has a `Tree`, but I haven't seen a `Graph` anywhere in Scalaland. It could be good to donate the code to either ScalaZ or Cats, or both.

Reply:
Perhaps would be good to create an issue in the Azavea 10% time repo pointing at the code and talking about potential contribution. There might be some folk (or yourself) that would want to take that on as a 10% time project.

Reply:
That's a good way to not forget about it, yeah. Doing so now.

Reply:
https://github.com/azavea/azavea-10-percent-time/issues/154